src/core/transport/upnp.cc

```diff
@@ -327,9 +322,14 @@ void UPnP::TryPortMapping(
             ":", strPort.c_str(), ")");
         return;
       }
-      std::this_thread::sleep_for(std::chrono::minutes(20));  // c++11
-      // boost::this_thread::sleep_for(); // pre c++11
-      // sleep(20*60); // non-portable
+      // TODO(unassigned): do we really want to retry on *all* errors? (see upnpcommands.h)
+      LogPrint(eLogError,
+          "UPnP: AddPortMapping (", strPort.c_str(),
+          ", ", strPort.c_str(),
+          ", ", m_NetworkAddr,
+          ") failed with code ", r);
+      // Try again later
+      std::this_thread::sleep_for(std::chrono::minutes(20));
```

Comment:
The magician works in mysterious ways.

I hoped that [this TODO](https://github.com/monero-project/kovri/pull/364/commits/150bbc7c1605080e3712ba81e8aab0661d38a7d5#diff-2faf3e582c3080ed0cd80f05dab8f1ceR57) would imply `upnp.cc:332` but we shouldn't take chances. I'll add the TODO.